lib/widgets/button.dart

```diff
@@ -0,0 +1,40 @@
+import 'package:flutter/material.dart';
+
+class Button extends StatelessWidget {
+  final String text;
+  final double? value;
+  final String color;
+  const Button({
+    Key? key,
+    required this.text,
+    this.value,
+    required this.color,
+  }) : super(key: key);
+
+  @override
+  Widget build(BuildContext context) {
+    final colorMap = {
+      'white': Colors.white,
+      'blue': Colors.blue,
+      'grey': Colors.grey,
+    };
+
+    final bgColor = colorMap[color] ?? Colors.grey;
```

Comment:
to używasz w jednym miejscu tylko, więc nie jest to potrzebne, implementacja w 28 l